Library/Homebrew/extend/os/linux/install.rb

```diff
@@ -5,21 +5,40 @@ module Homebrew
   module Install
     module_function
 
-    DYNAMIC_LINKERS = [
-      "/lib64/ld-linux-x86-64.so.2",
-      "/lib64/ld64.so.2",
-      "/lib/ld-linux.so.3",
-      "/lib/ld-linux.so.2",
-      "/lib/ld-linux-aarch64.so.1",
-      "/lib/ld-linux-armhf.so.3",
-      "/system/bin/linker64",
-      "/system/bin/linker",
+    DYNAMIC_LINKERS = %w[
+      /lib64/ld-linux-x86-64.so.2
+      /lib64/ld64.so.2
+      /lib/ld-linux.so.3
+      /lib/ld-linux.so.2
+      /lib/ld-linux-aarch64.so.1
+      /lib/ld-linux-armhf.so.3
+      /system/bin/linker64
+      /system/bin/linker
     ].freeze
     private_constant :DYNAMIC_LINKERS
 
+    GCC_VERSION_SUFFIX = OS::LINUX_GCC_CI_VERSION.sub("\.0", "").freeze
+
+    GCC_RUNTIME_LIBS = %w[
+      libgcc_s.so
+      libgcc_s.a
+      libgcc_s.so.1
+      libgomp.so
+      libgomp.a
+      libgomp.so.1
+      libquadmath.so
+      libquadmath.a
+      libquadmath.so.0
+      libstdc++.so
+      libstdc++.a
+      libstdc++.so.6
```

Comment:
```suggestion
      libgcc_s.so.1
      libgomp.so.1
      libquadmath.so.0
      libstdc++.so.6
```
The dynamic libraries with numeric suffixes are runtime requirements, namely `libgcc_s.so.1 libgomp.so.1 libquadmath.so.0 libstdc++.so.6`. The `*.so` and `*.a` files are build time requirements and not needed at run time. GCC is able to find its own libraries at build time, so they don't need to be symlinked into `$HOMEBREW_PREFIX/lib`. I tested removing these symlinks and building two C++ formulae `patchelf` (after removing the dependency on `gcc`) and `protobuf`, and both were successful.

Reply:
Nice, that will clean things up a bit here.